### PR TITLE
[#387] Do not escape the "timeout" parameter.

### LINKED ISSUE #387: [BUG] Heathcheck timeout parameter not parsable
### What is the bug?

the OpenSearch client cannot change the default timeout for health checks.

The cluster endpoint for healthcheck requires a string representing time units:

```
http "os-morel-1:9200/_cluster/health?timeout=4s"
```

It fails if you supply an integer or float.

```
(rayish) √ deus_lex % http "os-morel-1:9200/_cluster/health?timeout=4.2"
HTTP/1.1 400 Bad Request
content-encoding: gzip
content-length: 172
content-type: application/json; charset=UTF-8

{
    "error": {
        "reason": "failed to parse setting [timeout] with value [4.2] as a time value: unit is missing or unrecognized",
        "root_cause": [
            {
                "reason": "failed to parse setting [timeout] with value [4.2] as a time value: unit is missing or unrecognized",
                "type": "illegal_argument_exception"
            }
        ],
        "type": "illegal_argument_exception"
    },
    "status": 400
}
```

However, the client call requires a numeric value and fails with a time unit string

```
x = OpenSearch(["http://os-morel-1:9200"])

x.cluster.health(timeout="4s")

ConnectionError: ConnectionError(Timeout value connect was b'4s', but it must be an int, float or None.) caused by: ValueError(Timeout value connect was b'4s', but it must be an int, float or None.)

x.cluster.health(timeout=4)

ConnectionError: ConnectionError(Timeout value connect was 4, but it must be an int, float or None.) caused by: ValueError(Timeout value connect was 4, but it must be an int, float or None.)
```

### How can one reproduce the bug?
_Steps to reproduce the behavior._

### What is the expected behavior?

As with previous versions of elasticsearch client, I expect "timeout="4s" to work properly

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

### PR DESCRIPTION
**Pull request by grechaw**
When the "timeout" parameter is escaped and turned into a string, it does not pass validation checks.
This PR ensures that timeout is passed through as-is, just like "request_timeout".
There's no explicit test for this parameter, but I included it on some existing tests just to ensure that the code is exercised.  Without the fix in place, one of the tests as modified will not pass.

### Description
With this change, using the `timeout` parameter in `cluster.health` no longer throws a validation error.  I have not checked whether the problem exists elsewhere.

### Issues Resolved
Resolves #387 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by grechaw** [comment]
Based on feedback from my colleagues, I'll improve the testing.

**Review by dblock — CHANGES_REQUESTED** [review]
- add to CHANGELOG, this fixes a bug I think
- you mention there's no unit test - try adding one that is scoped to the `utils` implementation for this?

**Comment by grechaw** [comment]
Thanks for taking a look @dblock , I've added a simple test and I think this PR will help us move forward.  We have been moving to `opensearch-py` from a very old version of the Elasticsearch client.

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +71,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Updates `master` to `cluster_manager` to be inclusive ([#242](https://github.com/opensearch-project/opensearch-py/pull/242))
 - Support a custom signing service name for AWS SigV4 ([#268](https://github.com/opensearch-project/opensearch-py/pull/268))
 - Updated CI tests to make them work locally ([#275](https://github.com/opensearch-project/opensearch-py/pull/275))
+- Fix bug with validation of 'timeout' parameter ([#387](Do not escape the "timeout" parameter.))
```
Comment (dblock) [inline CHANGELOG.md:74]: Re-add the space ;)

**Review by dblock — APPROVED** [review]
Great. This is good with me, but since you're here, extend this test to the other parameters (ignore and request_timeout)? I want to make sure we don't introduce regressions. You'll have to fix `nox lint` as well, please.

**Comment by grechaw** [comment]
OK thanks again @dblock perhaps we are ready to go.   We all find the 'request_timeout' clobbering action odd, and I'm not sure what the 'ignore' parameter is for, but they're both under test now.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
LGTM! Thanks for hanging in here with me. 

Kicked integration tests again, we need to fix this flakiness ...

**Comment by grechaw** [comment]
It's my pleasure.  I will keep my eye out for more things to address.  Where possible it will be good to migrate more of our code to the `opensearch-py` implementation.    I'll look forward to the next release.

**Comment by dblock** [comment]
> It's my pleasure. I will keep my eye out for more things to address. Where possible it will be good to migrate more of our code to the `opensearch-py` implementation. I'll look forward to the next release.

If you want something meaty, I suggest https://github.com/opensearch-project/opensearch-py/issues/393 ;)

**Pull request merged**